test/e2e_test_cluster_image_policy.sh

```diff
@@ -544,3 +544,87 @@ echo '::group::' Cleanup
 kubectl delete cip --all
 kubectl delete ns demo-key-sha512
 echo '::endgroup::'
+
+# Publish the first test image
+echo '::group:: publish test image demoEphemeralImage'
+pushd $(mktemp -d)
+go mod init example.com/demo
+cat <<EOF > main.go
+package main
+import (
+  "fmt"
+  "time"
+)
+func main() {
+  // Calling Sleep method
+  time.Sleep(8 * time.Minute)
+
+  fmt.Println("Sleep Over.....")
+}
+EOF
+
+sed -i'' -e "s@TIMESTAMP@${TIMESTAMP}@g" main.go
+cat main.go
+export demoEphemeralImage=`ko publish -B example.com/demo`
+echo Created image $demoEphemeralImage
+popd
+echo '::endgroup::'
+
+echo '::group:: Deploy ClusterImagePolicy with keyless signing'
+kubectl apply -f ./test/testdata/policy-controller/e2e/cip-keyless.yaml
+echo '::endgroup::'
+
+echo '::group:: Sign demo image'
+if ! cosign sign --rekor-url ${REKOR_URL} --fulcio-url ${FULCIO_URL} --force --allow-insecure-registry ${demoEphemeralImage} --identity-token ${OIDC_TOKEN} ; then
+  echo "failed to sign with keyless"
+  exit 1
+fi
+echo '::endgroup::'
+
+echo '::group:: Verify demo image'
+if ! cosign verify --rekor-url ${REKOR_URL} --allow-insecure-registry ${demoEphemeralImage} ; then
+  echo "failed to verify with keyless"
+fi
+echo '::endgroup::'
+
+echo '::group:: Create test namespace and label for verification'
+export NS=demo-ephemeral-verification
+kubectl create namespace ${NS}
+kubectl label namespace ${NS} policy.sigstore.dev/include=true
+echo '::endgroup::'
+
+echo '::group:: test pod success'
+# We signed this above, this should work
+if ! kubectl run -n ${NS} poddemo --image=${demoEphemeralImage} ; then
+  echo Failed to create Pod in namespace with matching signature!
+  exit 1
+else
+  echo Succcessfully created Pod with signed image
+fi
+echo '::endgroup::'
+
+echo '::group:: Set no-match policy to deny'
+kubectl patch configmap/config-policy-controller \
+  --namespace cosign-system \
+  --type merge \
+  --patch '{"data":{"no-match-policy":"deny"}}'
+# allow for propagation
+sleep 10
+echo '::endgroup::'
+
+ephemeralContainerImage="busybox@sha256:9810966b5f712084ea05bf28fc8ba2c8fb110baa2531a10e2da52c1efc504698"
+
+echo '::group:: test rejection of ephemeral container that does not have any signature'
```

Comment:
Could we add a passing test case for the ephemeral container also? L598 creates the pod, but we don't seem to be (or I missed it), test the passing case?

Maybe just add:

```
if ! kubectl debug poddemo -n ${NS} --image=$(demoEphemeralContainerImage} ; then
  echo Failed to create a signed ephemeralcontainer ...
```

Reply:
yes

Reply:
This action reported an error when running Kubernetes 1.22, it makes sense as ephemeral containers were enabled by default in 1.23.